<?xml version="1.0"?>
<launch>
    <node pkg="urg_node" type="urg_node" name="scan_left">
        <param name="ip_address" value="192.168.0.10"/>
        <param name="frame_id" value="left_lidar_link"/>
        <param name="angle_max" value="2.355"/>
        <param name="angle_min" value="-2.355"/>
        <remap from="/scan" to="/scan_left"/>
    </node>

    <node pkg="urg_node" type="urg_node" name="scan_right">
        <param name="ip_address" value="192.168.5.99"/>
        <param name="frame_id" value="right_lidar_link"/>
        <param name="angle_max" value="2.355"/>
        <param name="angle_min" value="-2.355"/>
        <remap from="/scan" to="/scan_right"/>
    </node>

    <include file="$(find ira_laser_tools)/launch/laserscan_multi_merger.launch"/>
    
    <node name="serial_handler" pkg="stm32interface" type="serial_handler_node"/>


    <!-- <node pkg="laser_filters" type="scan_to_scan_filter_chain" name="laser_filter">
        <rosparam command="load" file="$(find slam_amcl)/config/LRF/laser_filter.yaml"/>
    </node> -->

    <node name="tf_repeater_node" pkg="sequence" type="tf_repeater_node">
        <rosparam command="load" file="$(find sequence)/config/LRF/amcl.yaml"/>
    </node>


    <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping">
        <rosparam command="load" file="$(find sequence)/config/LRF/gmapping.yaml"/>
        <remap from="/scan" to="/scan_filtered"/>
        <param name="base_frame" value="base_link"/>
        <param name="odom_frame" value="odom"/>
        <param name="map_frame" value="map"/>
    </node>

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find sequence)/rviz/gmapping.rviz" required="true"/>

</launch>